react-popup/my-app/src/Popup.tsx

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useRef } from 'react';
+import { createPortal } from 'react-dom';
+
+type Props = {
+  isOpen: boolean;
+  onClose: () => void;
+  positionTo: HTMLElement | null;
+  opacity?: number;
+  children: React.ReactNode;
+};
+
+export function Popup({
+  isOpen,
+  onClose,
+  positionTo,
+  opacity = 0.5,
+  children,
+}: Props) {
+  const popupRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
```

Comment:
With the onClick on the backdrop, I don't think you need the useEvent or event listeners.